cmd/oras/root/manifest/index/update.go

```diff
@@ -0,0 +1,235 @@
+/*
+Copyright The ORAS Authors.
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package index
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+
+	"github.com/opencontainers/go-digest"
+	ocispec "github.com/opencontainers/image-spec/specs-go/v1"
+	"github.com/spf13/cobra"
+	"oras.land/oras-go/v2"
+	"oras.land/oras-go/v2/content"
+	"oras.land/oras/cmd/oras/internal/argument"
+	"oras.land/oras/cmd/oras/internal/command"
+	"oras.land/oras/cmd/oras/internal/display/status"
+	oerrors "oras.land/oras/cmd/oras/internal/errors"
+	"oras.land/oras/cmd/oras/internal/option"
+	"oras.land/oras/cmd/oras/internal/output"
+	"oras.land/oras/internal/contentutil"
+	"oras.land/oras/internal/descriptor"
+)
+
+type updateOptions struct {
+	option.Common
+	option.Target
+
+	addArguments    []string
+	mergeArguments  []string
+	removeArguments []string
+	tags            []string
+}
+
+func updateCmd() *cobra.Command {
+	var opts updateOptions
+	cmd := &cobra.Command{
+		Use:   "update <name>{:<tag>|@<digest>} {--add/--merge/--remove} {<tag>|<digest>} [...]",
+		Short: "[Experimental] Update and push an image index",
+		Long: `[Experimental] Update and push an image index. All manifests should be in the same repository
+		
+Example - remove a manifest and add two manifests from an index tagged 'v1'. The tag will point to the updated index:
+  oras manifest index update localhost:5000/hello:v1 --add linux-amd64 --add linux-arm64 --remove sha256:99e4703fbf30916f549cd6bfa9cdbab614b5392fbe64fdee971359a77073cdf9
+
+Example - update an index by specifying its digest:
```

Comment:
Unlike updating the index with a tag, it actually creates a new index with a different digest.
```suggestion
Example - Create a new index by updating an existing index specified by its digest:
```